### PR TITLE
Add skeleton structure with endpoints

### PR DESCRIPTION
**Pull request by jphillips2121**
- Fixed Makefile 
- Added Config package for app
- Added go mod files 
- Added two empty endpoints for returning company directors, and creating auth code request
- Taken Kafka initialising out of Start script as not being used here.

Resolves: BI-3362

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+	"github.com/companieshouse/gofigure"
+)
+
+var cfg *Config
+var mtx sync.Mutex
+
+// Config defines the configuration options for this service.
+type Config struct {
+	BindAddr string `env:"BIND_ADDR"                      flag:"bind-addr"                       flagDesc:"Bind address"`
```
Comment (emead) [inline config/config.go:15]: Indentation could be reduced

File: Makefile [hunk 2]
```diff
@@ -22,17 +22,17 @@ fmt:
...
-	go build
+	CGO_ENABLED=0 go build -o ./$(bin)
 
 .PHONY: test
 test: test-unit test-integration
 
 .PHONY: test-unit
 test-unit:
-	go test $(TESTS) -run 'Unit'
+	go test $(TESTS) -run 'Unit' -coverprofile=coverage.out
```
Comment (emead) [inline Makefile:35]: Some of these changes don't match the styleguides.
Do the styleguides need updating?
https://github.com/companieshouse/styleguides/blob/master/Makefiles/go_makefiles.md

File: handlers/register.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"net/http"
+
+	"github.com/companieshouse/chs.go/authentication"
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/emergency-auth-code-api/config"
+	"github.com/gorilla/mux"
+)
+
+// Contains config and DAO for
```
Comment (emead) [inline handlers/register.go:12]: Comment should begin with name
```suggestion
// EmergencyAuthCodeService contains config and DAO for
```

File: handlers/register.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+type EmergencyAuthCodeService struct {
+	Config *config.Config
+}
+
```
Comment (emead) [inline handlers/register.go:16]: Exported function needs a comment
```suggestion
// Register defines the endpoints for the API
```

File: Makefile (hunk 2 above)
Comment (jphillips2121) [reply Makefile:35]: Hadn't seen that, I'll revert the changes. 

`-coverprofile=coverage.out` displays unit test coverage for each file it's run against, so I think that should stay as it's useful to see. I'd suggest the styleguide would need updating with that. 

I also get an error when leaving `deps` in the build command on line 28 so decided to take it out, I can't seem to find it in any other go repos we use

**Pull request merged**